regclient/regclient
# Removing external imports of image schemas

Signed-off-by: Brandon Mitchell <git@bmitch.net>



### Fixes issue



### Describe the change

This transitions from external to internal types for the various json schemas of manifests/configs/etc. One of the more user visible changes will be the descriptor moving from OCI's image-spec to the types package. This change is being performed to allow features to be added to these types ahead of being approved upstream. This also simplifies many of the backend processes that were converting between types since there's now a single platform and descriptor type across the various schemas.

### How to verify it

Various type changes will break existing code using these APIs. Once code is updated to use the new types, all other changes should be transparent.

### Changelog text



### Please verify and check that the pull request fulfills the following requirements
